node/src/Unison/Runtime/Cryptography.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# Language ScopedTypeVariables #-}
+
+module Unison.Runtime.Cryptography where
+
+import Unison.Cryptography
+import Data.ByteString (ByteString)
+import Data.ByteArray as BA
+import qualified Crypto.Random as R
+
+-- cryptonite
+-- import Crypto.Cipher.Types.Base ()
+-- import Crypto.Cipher.AES.Primitive (AES)
+import qualified Crypto.Cipher.AES as AES
+import Crypto.Cipher.Types
+import Crypto.Error
+
+-- Creates a Unison.Cryptography object specialized to use the noise protocol
+-- (http://noiseprotocol.org/noise.html).
+mkCrypto :: forall cleartext symmetricKey . (ByteArrayAccess cleartext, ByteArray cleartext, ByteArrayAccess symmetricKey, ByteArray symmetricKey) => ByteString -> Cryptography ByteString symmetricKey () () () () cleartext
```

Comment:
`symmetricKey` should be concrete, based on the cipher used. Suggest defining a `newtype SymmetricKey = AES256 ByteString`. Give a `ByteArrayAccess SymmetricKey`. Don't export the constructor, provide a function `symmetricKey :: ByteString -> Maybe SymmetricKey` which fails if the key isn't a valid AES256 (or whatever key type).